xrpl/models/utils.py

```diff
@@ -1,9 +1,41 @@
 """Helper util functions for the models module."""
+
+import re
 from dataclasses import dataclass, is_dataclass
-from typing import Any, Dict, List, Type, TypeVar, cast
+from typing import Any, Dict, List, Optional, Pattern, Type, TypeVar, cast
+
+from typing_extensions import Final
 
 from xrpl.models.exceptions import XRPLModelException
 
+HEX_REGEX: Final[Pattern[str]] = re.compile("[a-fA-F0-9]*")
+
+# Credentials are represented in hex. Whilst they are allowed a maximum length of 64
+# bytes, every byte requires 2 hex characters for representation
+_MAX_CREDENTIAL_LENGTH: Final[int] = 64 * 2
+
+
+def validate_credential_ids(credential_list: Optional[List[str]]) -> Dict[str, str]:
+    """
+    Returns:
+        Errors pertaining to credential_ids field
+
+    Args:
+        credential_list: An optional list of input credentials
+
+    """
+    errors = {}
+    if credential_list is not None:
+        if len(credential_list) == 0:
+            errors["credential_ids"] = "CredentialIDs list cannot be empty."
+        if len(credential_list) > 8:
+            errors[
+                "credential_ids"
+            ] = "CredentialIDs list cannot have more than 8 elements."
```

Comment:
Need a check here to ensure no duplicates

Reply:
a6ac0f47